bokeh/models/annotations.py

```diff
@@ -103,33 +157,72 @@ class Legend(Annotation):
     The width (in pixels) that the rendered legend glyph should occupy.
     """)
 
-    legend_margin = Int(10, help="""
+    margin = Int(10, help="""
     Amount of margin around the legend.
     """)
 
-    legend_padding = Int(10, help="""
+    padding = Int(10, help="""
     Amount of padding around the contents of the legend.
     """)
 
-    legend_spacing = Int(3, help="""
+    spacing = Int(3, help="""
     Amount of spacing between legend entries.
     """)
 
-    legends = Either(List(Instance(GlyphRenderer)), List(Tuple(String, List(Instance(GlyphRenderer)))), help="""
-    A list of glyph renderers OR
-    A list of tuples that maps text labels to the legend to corresponding
-    renderers that should draw sample representations for those labels.
-
-    If a list of glyph renderers is provided, the label property from the glyph
-    will be used as the legend label.
-
-    .. note::
-        The ``legends`` attribute may also be set from a dict or OrderedDict.
-        If a dict is used, the order of the legend entries is unspecified.
+    items = List(Instance(LegendItem), help="""
+    A list of legend items to be rendered in the legend.
+    """)
+
+    @property
+    def legends(self):
+        warnings.warn(DEP_MSG_0_12_3 % ('legends', 'Legend.items'))
```

Comment:
You have to use `BokehDeprecationWarning`. Otherwise most users won't see this. Apply in other instances as well.